botocore/endpoint.py

```diff
@@ -152,16 +130,13 @@ def _create_request_object(self, request_dict):
                              headers=headers)
         return request
 
-    def prepare_request(self, request, signer):
-        if signer is not None:
-            with self._lock:
```

Comment:
I believe this was only needed for the `fix_s3_host` customization. It was updated to be thread-safe, and now we should no longer need a lock. Maybe @jamesls can elaborate?